connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2576,11 +2578,12 @@ class SparkConnectPlanner(val session: SparkSession) {
 
       case StreamingQueryCommand.CommandCase.AWAIT_TERMINATION =>
         if (command.getAwaitTermination.hasTimeoutMs) {
-          val terminated = query.awaitTermination(command.getAwaitTermination.getTimeoutMs)
+          val terminated = handleStreamingAwaitTermination(query,
+            Some(command.getAwaitTermination.getTimeoutMs))
           respBuilder.getAwaitTerminationBuilder
             .setTerminated(terminated)
         } else {
-          query.awaitTermination()
+          handleStreamingAwaitTermination(query, None)
           respBuilder.getAwaitTerminationBuilder
             .setTerminated(true)
         }
```

Comment:
Can we simplify this? We don't need larger conditional.  Could just be:

```scala
val timeout = if (command.getAwaitTermination.hasTimeoutMs)
   Some(command.getAwaitTermination.getTimeoutMs) else None
val terminated = handleStreamingAwaitTermination(query, timeout)
  respBuilder.getAwaitTerminationBuilder
  .setTerminated(terminated)

Reply:
done